evap/staff/views.py

```diff
@@ -677,48 +678,91 @@ def course_participant_import(request, semester_id, course_id):
     if request.method == "POST":
         operation = request.POST.get('operation')
 
-        if operation not in ('test', 'import', 'copy'):
+        if operation not in ('test-participants', 'import-participants', 'copy-participants',
+                             'test-contributors', 'import-contributors', 'copy-contributors'):
             raise SuspiciousOperation("Invalid POST operation")
 
-        test_run = operation == 'test'
-        import_run = operation == 'import'
-        copy_run = operation == 'copy'
+        if 'participants' in operation:
+            excel_form = participant_excel_form
+            copy_form = participant_copy_form
+        else:                                                 # 'contributors' in operation
+            excel_form = contributor_excel_form
+            copy_form = contributor_copy_form
 
-        if test_run:
-            delete_import_file(request.user.id)  # remove old file if still exists
-            excel_form.excel_file_required = True
-            if excel_form.is_valid():
-                excel_file = excel_form.cleaned_data['excel_file']
-                file_content = excel_file.read()
-                # if on a test run, the process method will not return a list of users that would be
-                # imported so there is currently no way to show how many users would be imported.
-                __, success_messages, warnings, errors = UserImporter.process(file_content, test_run)
-                if not errors:
-                    save_import_file(excel_file, request.user.id)
+        result, success_messages, warnings, errors = helper_course_person_import(
+                request, excel_form, copy_form, course, semester_id)
```

Comment:
i think i would inline that method again. it would be a different thing if we had two of these methods, but right now it's just one. of course the view gets quite massive, but now it makes things even worse by e.g. making that redirect weirdness necessary.